ocs_ci/ocs/node.py

```diff
@@ -69,26 +69,24 @@ def wait_for_nodes_status(node_names=None, status=constants.NODE_READY, timeout=
         timeout (int): The number in seconds to wait for the nodes to reach
             the status
 
-    Returns:
-        bool: True if all nodes reached the status, False otherwise
-
     """
     if not node_names:
         node_names = [node.name for node in get_node_objs()]
 
     log.info(f"Waiting for nodes {node_names} to reach status {status}")
-
     try:
         for sample in TimeoutSampler(timeout, 3, get_node_objs, node_names):
             for node in sample:
-                if not node_names:
-                    return True
                 if node.ocp.get_resource_status(node.name) == status:
                     node_names.remove(node.name)
-                    break
+            if not node_names:
+                break
+
     except TimeoutExpiredError:
         log.error(f"The following nodes haven't reached status {status}: {node_names}")
-        return False
+        raise exceptions.ResourceWrongStatusException(
```

Comment:
The log error should be in the raise imo cause when we sill see test fail, we won't see the error but just in the output console